### ISSUE TITLE
Multi-Root: Configure projects with Open Folder without a workspace (automatically)

### ISSUE THREAD

**Issue by Zingam** [microsoft/vscode-cmake-tools#1024]
Currently as I see it to use the Multi-Root feature. I have to use the option "Add Folder to Workspace" and have a workspace.

I'd like to be able to just point to a root folder containing sub folders with independent CMakeList.txt projects and the extension to find them and configure them automatically without the hassle to create a workspace and add the folders one by one.

```
MyProjects/
     MyProject01/CMakeLists.txt
     MyProject02/CMakeLists.txt
     MyProject03/CMakeLists.txt
```
In that context "cmake.sourceDirectory" could point to an array of root CMakeLists.txt files

**Comment by Zingam** [comment]
Or alternatively `.vscode` in the root folder could contain a common and also a separate configuration for each project.

**Comment by bobbrow** [comment]
> Currently as I see it to use the Multi-Root feature. I have to use the option "Add Folder to Workspace" and have a workspace

You are correct.  There are two ways to configure multiple projects.  We did the first one, and changing the `cmake.sourceDirectory` to accept an array of strings would be the way to support the other one.

**Comment by withattribution** [comment]
TLDR: +1 array of strings for sourceDirectory

This would also help my situation where I have sibling folders each with their own projects as in

```
Application/                 — Application project directory
  + Makefile / CMakeLists.txt    - Makefiles of the application project
  + build/
   - components/          — Components of the application project
     + testable/
  + main/                       — Main source files of the application project
  - test/                         — Test project directory
     + build/
     + Makefile / CMakeLists.txt    - Makefiles of the test project

```
Which is typical of how unit testing is done with [espressif eps32](https://github.com/espressif/esp-idf/tree/master/examples/system/unit_test)

As a workaround I have "re-added" the test folder to create a workspace so that CMakeTools sees the separate $workspaceFolder's and allows me to build the two projects given their independent root CMakeLists.

While it's confusing to have the same folder represented twice (test folder as a root folder and test folder in the main application's root folder) -- it's worth it to get rid of the distracting red squiggles!
